token-swap/program/src/curve/constant_price.rs

```diff
@@ -105,12 +122,21 @@ impl CurveCalculator for ConstantPriceCurve {
             .checked_mul(token_b_price)?
             .checked_add(U256::from(swap_token_a_amount))?;
         let pool_supply = U256::from(pool_supply);
-        Some(
-            pool_supply
-                .checked_mul(given_value)?
-                .checked_div(total_value)?
-                .as_u128(),
-        )
+        match round_direction {
+            RoundDirection::Floor => Some(
+                pool_supply
+                    .checked_mul(given_value)?
+                    .checked_div(total_value)?
+                    .as_u128(),
+            ),
+            RoundDirection::Ceiling => Some(
+                pool_supply
+                    .checked_mul(given_value)?
+                    .checked_add(U256::from(1))?
```

Comment:
I'll add ceiling division for U256

Otherwise, there's `check_pool_token_conversion` which does a few round trips to make sure no value is lost, but I can add another one which just checks that no value is lost on one `trading_tokens_to_pool_tokens`.

Reply:
Up to you, this math just didn't look quite right so I figured there might be some missing test coverage

Reply:
Done now with `ee618f2`